crates/tx5-connection/src/lib.rs

```diff
@@ -66,20 +95,23 @@ impl<T: 'static + Send> Clone for CloseSend<T> {
 impl<T: 'static + Send> Drop for CloseSend<T> {
     fn drop(&mut self) {
         if self.close_on_drop {
-            self.sender.lock().unwrap().take();
+            let s = self.sender.lock().unwrap().take();
+            if let Some(mut s) = s {
+                s.close_channel();
```

Comment:
This switch to a futures::channel instead of a tokio lets us explicitly close the channel from the sender side. It turned out this fix is not technically needed, but I left it in because it's more explicit about our intentions here.